device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/DeviceClientTest.java

```diff
@@ -1988,6 +1988,364 @@ public void setCertificateAuthorityThrowsIfOpenTransportClient()
 
     }
 
+    @Test
+    public void setAmqpOpenAuthenticationSessionTimeout()
+    {
+        //arrange
+        final int timeoutInSeconds = 10;
+        DeviceClient client = Deencapsulation.newInstance(DeviceClient.class);
+        Deencapsulation.setField(client, "config", mockConfig);
+
+        new NonStrictExpectations()
+        {
+            {
+                mockConfig.getProtocol();
+                result = IotHubClientProtocol.AMQPS;
+
+                mockConfig.getAuthenticationType();
+                result = DeviceClientConfig.AuthType.SAS_TOKEN;
+            }
+        };
+
+        //act
+        client.setOption("SetAmqpOpenAuthenticationSessionTimeout", timeoutInSeconds);
+
+        //assert
+        new Verifications() {
+            {
+                Deencapsulation.invoke(mockConfig, "setAmqpOpenAuthenticationSessionTimeout", timeoutInSeconds);
+                times = 1;
+            }
+        };
+    }
+
+    @Test
+    public void setAmqpWsOpenAuthenticationSessionTimeout()
+    {
+        //arrange
+        final int timeoutInSeconds = 10;
+        DeviceClient client = Deencapsulation.newInstance(DeviceClient.class);
+        Deencapsulation.setField(client, "config", mockConfig);
+
+        new NonStrictExpectations()
+        {
+            {
+                mockConfig.getProtocol();
+                result = IotHubClientProtocol.AMQPS_WS;
+
+                mockConfig.getAuthenticationType();
+                result = DeviceClientConfig.AuthType.SAS_TOKEN;
+            }
+        };
+
+        //act
+        client.setOption("SetAmqpOpenAuthenticationSessionTimeout", timeoutInSeconds);
+
+        //assert
+        new Verifications() {
+            {
+                Deencapsulation.invoke(mockConfig, "setAmqpOpenAuthenticationSessionTimeout", timeoutInSeconds);
+                times = 1;
+            }
+        };
+    }
+
+    @Test
+    public void setNullAmqpWsOpenAuthenticationSessionTimeoutThrows()
+    {
+        //arrange
+        DeviceClient client = Deencapsulation.newInstance(DeviceClient.class);
+        Deencapsulation.setField(client, "config", mockConfig);
+
+        try {
+            //act
+            client.setOption("setAmqpOpenAuthenticationSessionTimeout", null);
+        } catch (IllegalArgumentException expected) {
```

Comment:
This seemed like the best to validate that the correct exception is thrown. The exception type itself can be thrown for various reasons but we want to ensure it is thrown from the correct part of code.